docs: shells for declarative package managers (guix and nix)

pelasgus opened the pull request:
Concerning:
- https://github.com/tauri-apps/wry/pull/1377
- https://github.com/tauri-apps/wry/pull/1373
- https://github.com/tauri-apps/wry/pull/990

> [!NOTE]
> **N.B.**: Im willing to transfer the repo to the organisation for easier management and future updates and maintenance. Its easier overall for it to be a submodule.

> [!IMPORTANT]
> **P.S.**: Will be adding a flake for `nix` as well.

@amrbashir was the original reviewer. please when you have a minute, check this out.

pelasgus committed:
Declerative Shells

Added development shells for declerative package managers guix and nix.

pelasgus committed:
Documentation

pelasgus commented:
If you do take ownership of the repository, then we could perhaps make the shells a little bit more universal so that they could be employed by all tauri repos; thus making development much easier. Otherwise, I can restructure the pr so that instead of a sub-module I add the scripts directly(specialised) to each repo.

amrbashir commented on README.md:
````diff
@@ -83,7 +83,8 @@ Wry also needs [WebKitGTK](https://webkitgtk.org/) for WebView. So please make s
 ```
 
-The `libayatana-indicator` package can be installed from the Arch User Repository (AUR).
+> [!IMPORTANT]
````
let's remove this info, it is outdated and not a concern for this repo anymore

amrbashir commented on README.md:
````diff
@@ -97,7 +98,53 @@ sudo apt install libwebkit2gtk-4.1-dev
 ```
 
-Fedora does not have the Ayatana package yet, so you need to use the GTK one, see the [feature flags documentation](https://docs.rs/wry/latest/wry/#feature-flags).
+> [!WARNING]
````
ditto, this info is no longer needed and should be removed

amrbashir commented on README.md:
```diff
@@ -97,7 +98,53 @@ sudo apt install libwebkit2gtk-4.1-dev
+    packages = with pkgs; [
+      pkg-config
+      webkitgtk_4_1
+      libayatana-appindicator
+      libappindicator-gtk3
```
these two packages are no longer a concern for this repo, we should remove them from here

amrbashir commented on README.md:
````diff
@@ -97,7 +98,53 @@ sudo apt install libwebkit2gtk-4.1-dev
+  }
+ ```
+
+ > [!TIP]
+ > A `shell.nix` is included in the subdirectory `declerative-shells/etc/shells/nix`. Use as follows:
+ ```sh
+ nix-shell shell.nix
+ ```
+
+#### GUIX
+
+ ```scheme
+ (specifications->manifest
+   '("pkg-config"                ; Helper tool used when compiling
+     "webkitgtk"                 ; Web content engine fot GTK+
+     "libappindicator"           ; Menu in Menu bar
+     "gtk"
+  ))
+ ```
+
+ > [!TIP]
+ > A `manifest.scm` is included in the subdirectory `declerative-shells/etc/shells/guix`. Use as follows:
+ 
+```sh
+ guix shell -m manifest.scm
+ ```
+
+ > [!WARNING]
+ > Guix does not have the Ayatana package yet, so you need to use the GTK one, see the [feature flags documentation](https://docs.rs/wry/latest/wry/#feature-flags).
````
This is not needed and the submodule as well, the instructions at Line 104 are enough

pelasgus committed:
Applied Requested Changes

Removed some unnecessary packages and mentions to them. Removed git submodule to declerative dev shells and sorted the documentation so that they are in-line.

pelasgus committed:
Applied Requested changes

Correctly removed git submodule.

pelasgus committed:
Typo

pelasgus committed:
Typo

amrbashir commented on README.md:
````diff
@@ -109,6 +140,9 @@ If you are cross-compiling for macOS using [osxcross](https://github.com/tpoecht
 RUSTFLAGS="-l framework=WebKit" cargo build --target=x86_64-apple-darwin --release
 ```
 
+> [!IMPORTANT]
+> If `nix` and/if `nix-darwin` are used then follow instructions from [Nix Section](#nix--nixos).
````
Nix section uses webkit2gtk where on macOS we use the included WKWebView, let's remove this

pelasgus committed:
Merge branch 'tauri-apps:dev' into pelasgus/declerative-shells

pelasgus committed:
Implemented Requested Changes
